### PR TITLE
deps: remove aiofiles

### LINKED ISSUE #5306: dependencies: drop aiofiles
Aiofiles is a convenience library for farming out `os` calls to an executor (i.e. running them in threads).

The library essentially comes down to this 8 line snippet.

https://github.com/Tinche/aiofiles/blob/ee14e26f1cfee8b39888e7a03e70fba984d8845f/src/aiofiles/os.py#L7-L15

We don't really need this dependency as rolling your own async wraper is trivial. They've recently moved to date based versioning which pushes the overheads of compatibility onto us much more than before. This given the immaturity of the library means there's a reasonable chance that maintaining it as a dependency may become hard work at our end.

**Pull requests welcome!**

### PR DESCRIPTION
**Pull request by oliver-sanders**
* Aiofiles is a simple wrapper for a thread pool executor. We don't need a 3rd party library to do this for us.
* Closes #5306

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [x] Tests are included (or explain why tests are not needed).
- [x] `CHANGES.md` entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.
- [ ] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

**Review by MetRonnie — APPROVED** [review]
LGTM. I have done a quick test of playing + cleaning a workflow and running the fast tests locally

File: cylc/flow/workflow_files.py [hunk 1]
```diff
@@ -640,29 +648,7 @@ def load_contact_file(reg: str) -> Dict[str, str]:
...
-        async with aiofiles.open(path, mode='r') as cont:
-            data = {}
-            async for line in cont:
-                key, value = [item.strip() for item in line.split("=", 1)]
-                # BACK COMPAT: contact pre "suite" to "workflow" conversion.
-                # from:
-                #     Cylc 8
-                # remove at:
-                #     Cylc 8.x
-                data[key.replace('SUITE', 'WORKFLOW')] = value
```
Comment (datamel) [inline cylc/flow/workflow_files.py:662]: Will removing this catch anyone out?

Comment (MetRonnie) [reply cylc/flow/workflow_files.py:662]: It was a duplicate of what was already in `load_contact_file()` so not actually removed

Comment (oliver-sanders) [reply cylc/flow/workflow_files.py:662]: I've replaced this with the sync `load_contact_file` interface run inside a thread via `make_async`.

**Review by datamel — APPROVED** [review]

**Comment by datamel** [comment]
Flaky tests failing with a codecov upload so unrelated to this pr.

**Pull request merged**